### DISCUSSION TITLE
Is it possible to change the note template based on condition?

### DISCUSSION THREAD

**Discussion by onnyyonn** [emacs-citar/citar#472] [Q&A] [answered]
Some of my references have the doi field while the others have the url field. Is it possible to include url or doi in the template only when those fields exist?

**Comment by bdarcus** [comment, marked as answer]
Can't you just do `${doi url}`? That should only print the first one found, or none.

If not (for example, you want to prepend "doi:" if using the doi), then you're only alternative it to write your own format-note function, based on the default one.

Or see #439 if you have thoughts on that effort.

**Reply by onnyyonn** [reply]
Thanks. Using `${doi url}` is enough for my needs for now.
